src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -261,96 +262,184 @@ public ExamScoresDTO calculateExamScores(Long examId) {
+     * Includes the corresponding grade and grade type as well if a GradingScale is set for the relevant exam.
+     * @param studentExam a StudentExam instance that will have its points and grades calculated if it is assessed
+     * @param participationsOfStudent StudentParticipation list for the given studentExam
+     * @return Student Exam results with exam grade calculated if applicable
+     */
+    @NotNull
+    public StudentExamWithGradeDTO calculateStudentResultWithGradeAndPoints(StudentExam studentExam, List<StudentParticipation> participationsOfStudent) {
+        Exam exam = studentExam.getExam();
+
+        // Adding exam information to DTO
+        var scores = new ExamScoresDTO(exam.getId(), exam.getTitle(), exam.getMaxPoints());
+        var objectMapper = new ObjectMapper();
+
+        Optional<GradingScale> gradingScale = gradingScaleRepository.findByExamId(exam.getId());
+        ExamScoresDTO.StudentResult studentResult = calculateStudentResultWithGrade(studentExam, participationsOfStudent, exam, scores, objectMapper, gradingScale, false);
+        var studentExamWithGradeDTO = new StudentExamWithGradeDTO(studentExam, studentResult);
+
+        gradingScale.ifPresent(scale -> studentExamWithGradeDTO.gradeType = scale.getGradeType());
+
+        List<Exercise> exercises = studentExam.getExercises();
+        studentExamWithGradeDTO.maxPoints = calculateMaxPointsSum(exercises, exam.getCourse());
+        studentExamWithGradeDTO.maxBonusPoints = calculateMaxBonusPointsSum(exercises, exam.getCourse());
+        studentExamWithGradeDTO.achievedPointsPerExercise = calculateAchievedPointsForExercises(participationsOfStudent, exam.getCourse());
+
+        return studentExamWithGradeDTO;
+    }
+
+    @NotNull
+    private ExamScoresDTO.StudentResult calculateStudentResultWithGrade(StudentExam studentExam, List<StudentParticipation> participationsOfStudent, Exam exam,
+            ExamScoresDTO scores, ObjectMapper objectMapper, Optional<GradingScale> gradingScale, boolean calculateFirstCorrectionPoints) {
+        User user = studentExam.getUser();
+        var studentResult = new ExamScoresDTO.StudentResult(user.getId(), user.getName(), user.getEmail(), user.getLogin(), user.getRegistrationNumber(),
+                studentExam.isSubmitted());
+
+        studentResult.overallPointsAchieved = 0.0;
+        studentResult.overallPointsAchievedInFirstCorrection = 0.0;
+        for (StudentParticipation studentParticipation : participationsOfStudent) {
+            Exercise exercise = studentParticipation.getExercise();
+
+            // Relevant Result is already calculated
+            if (studentParticipation.getResults() != null && !studentParticipation.getResults().isEmpty()) {
+                Result relevantResult = studentParticipation.getResults().iterator().next();
+                // Note: It is important that we round on the individual exercise level first and then sum up.
```

Comment:
Optional:
I don't really think this comment still makes sense here, as the `calculateAchievedPoints` method name does not really indicate that it also rounds the score. I think you should either specify this as a JavaDoc comment for the `calculateAchievedPoints` method or rename it so it still contains the keyword "round", as it was before.

Reply:
I added the comment to the documentation and mentioned rounding as well, I haven't included it in the name because rounding can be considered a part of the internal calculation logic and caller shouldn't depend on that in case it changes later, and also to be consistent we would need to add "round" keyword to all related methods. I am not sure extra verbosity will be significantly helpful to the caller.